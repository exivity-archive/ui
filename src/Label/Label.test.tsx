import React from 'react'
import renderer from 'react-test-renderer'
import { Label } from '.'

test('renders label without props', () => {
  const button = renderer.create(<Label />)
  expect(button.toJSON()).toMatchSnapshot()
})

test('renders label with text', () => {
  const button = renderer.create(<Label>label</Label>)
  expect(button.toJSON()).toMatchSnapshot()
})

test('renders secondary label', () => {
  const button = renderer.create(<Label secondary>secondary</Label>)
  expect(button.toJSON()).toMatchSnapshot()
})

test('renders nested label', () => {
  const button = renderer.create(<Label>primary<Label>secondary</Label></Label>)
  expect(button.toJSON()).toMatchSnapshot()
})
