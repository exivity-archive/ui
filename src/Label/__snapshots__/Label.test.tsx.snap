// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders label with text 1`] = `
.c1 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c1 > .c0 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

<label
  className="c0 c1"
>
  label
</label>
`;

exports[`renders label without props 1`] = `
.c1 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c1 > .c0 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

<label
  className="c0 c1"
/>
`;

exports[`renders nested label 1`] = `
.c1 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
}

.c1 > .c0 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c2 > .c0 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

<label
  className="c0 c1"
>
  primary
  <label
    className="c0 c1"
  >
    secondary
  </label>
</label>
`;

exports[`renders secondary label 1`] = `
.c2 > .c0 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c1 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  display: block;
  font-size: 0.8125em;
  color: #999;
}

.c1 > .c0 {
  display: block;
  font-size: 0.8125em;
  color: #999;
}

<label
  className="c0 c1"
>
  secondary
</label>
`;
